src/app/ideastable/ideas-table.component.html

```diff
@@ -0,0 +1,21 @@
+<div *ngIf="dataLength">
+  <table mat-table [dataSource]="dataSource" class="mat-elevation-z8">
```

Comment:
dataLength and dataSource are quite vague, maybe better to rename them to something like "ideasLength", "ideasSource"